deepcell_spots/applications/polaris.py

```diff
@@ -436,19 +436,25 @@ def _predict(self,
             print('Refining spot locations.')
             dec_prob_im = np.zeros((spots_image.shape[:3]))
 
-            for i in range(len(df_results)):
-                gene = df_results.loc[i, 'predicted_name']
-                if gene in ['Background', 'Unknown']:
-                    continue
-                if 'Blank' in gene:
-                    continue
-                
-                x = df_results.loc[i, 'x']
-                y = df_results.loc[i, 'y']
-                b = df_results.loc[i, 'batch_id']
-                prob = max_proj_images[b, x, y]
-                
-                dec_prob_im[b, x, y] = prob
+            # Mask out unwanted genes
+            mask_valid = ~df_results['predicted_name'].isin(['Background', 'Unknown']) & \
+                         ~df_results['predicted_name'].str.contains('Blank', na=False)
+            valid_df = df_results[mask_valid]
+
+            b = valid_df['batch_id'].to_numpy()
+            x = valid_df['x'].to_numpy()
+            y = valid_df['y'].to_numpy()
+
+            dec_prob_im[b, x, y] = max_proj_images[b, x, y]
+
+
+            # Pre-create a dictionary for fast lookups
+            lookup_dict = {}
+            for idx, row in df_results.iterrows():
+                key = (row['batch_id'], row['x'], row['y'])
+                if key not in lookup_dict:
+                    lookup_dict[key] = []
+                lookup_dict[key].append(idx)
```

Comment:
This shouldn't affect performance at all, but one cool utility that I'm constantly using for situations like this is [`defaultdict`](https://docs.python.org/3/library/collections.html#collections.defaultdict). It essentially allows you to skip the `if not in` lookup as the default value for missing keys can be made an empty list (or set, or other container, etc.). So this could be something like (moving the import to a more appropriate location):

```suggestion
            # Pre-create a dictionary for fast lookups
            from collections import defaultdict
            lookup_dict = defaultdict(list)
            for idx, row in df_results.iterrows():
                lookup_dict[(row['batch_id'], row['x'], row['y'])].append(idx)
```